haskell-src/Concordium/Crypto/ByteStringHelpers.hs

```diff
@@ -125,9 +126,9 @@ instance AE.ToJSON ByteStringHex where
 -- |JSON instances based on base16 encoding.
 instance AE.FromJSON ByteStringHex where
   parseJSON = AE.withText "ByteStringHex" $ \t ->
-    let (bs, rest) = BS16.decode (Text.encodeUtf8 t)
-    in if BS.null rest then return (ByteStringHex (BSS.toShort bs))
-       else AE.typeMismatch "Not a valid Base16 encoding." (AE.String t)
+    case BS16.decode (Text.encodeUtf8 t) of
+      Right bs -> return (ByteStringHex (BSS.toShort bs))
+      Left _ -> AE.typeMismatch "Not a valid Base16 encoding." (AE.String t)
```

Comment:
For debugging it would be useful to keep the error.

Reply:
when needed it can be changed